### PR TITLE
add support for trimesh collider

### PR DESCRIPTION
**Pull request by dinomut1**

### CODE REVIEW

File: packages/engine/src/physics/classes/Physics.ts [hunk 1]
```diff
@@ -151,7 +151,7 @@ function createColliderDesc(
   // if (!colliderDescOptions.shapeType && mesh.geometry.type === 'CylinderGeometry')
   //   colliderDescOptions.shapeType = ShapeType.Cylinder
 
-  if (typeof colliderDescOptions.shapeType === 'undefined') return
+  if (!overrideShapeType && typeof colliderDescOptions.shapeType === 'undefined') return
```
Comment (HexaField) [inline packages/engine/src/physics/classes/Physics.ts:154]: is this necessary if we are guaranteed to be passing shapeType in?

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/engine/src/scene/components/ColliderComponent.ts [hunk 2]
```diff
@@ -108,8 +108,9 @@ export const ColliderComponent = defineComponent({
     const isLoadedFromGLTF = useOptionalComponent(entity, GLTFLoadedComponent)
     const groupComponent = useOptionalComponent(entity, GroupComponent)
 
+    const isMeshCollider = [ShapeType.TriMesh, ShapeType.ConvexPolyhedron].includes(colliderComponent.shapeType.value)
```
Comment (HexaField) [inline packages/engine/src/scene/components/ColliderComponent.ts:111]: shouldn't we also be checking for ModelComponent here?

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/engine/src/physics/classes/Physics.ts (hunk 1 above)
Comment (dinomut1) [reply packages/engine/src/physics/classes/Physics.ts:154]: We are not guaranteed to be passing shapeType in. ColliderDescOptions defines shapeType as possibly undefined

File: packages/engine/src/scene/components/ColliderComponent.ts (hunk 2 above)
Comment (dinomut1) [reply packages/engine/src/scene/components/ColliderComponent.ts:111]: We don't use ModelComponent at any point, just the GroupComponent

Comment (dinomut1) [reply packages/engine/src/scene/components/ColliderComponent.ts:111]: We should leave the door open to allow any kind of component to supply the mesh that is used

**Pull request merged**